plugins/services/src/js/components/modals/EditServiceModal.js

```diff
@@ -6,17 +6,33 @@ import CreateServiceModal
   from "#PLUGINS/services/src/js/components/modals/CreateServiceModal";
 import DCOSStore from "#SRC/js/stores/DCOSStore";
 import FullScreenModal from "#SRC/js/components/modals/FullScreenModal";
+import RequestErrorMsg from "#SRC/js/components/RequestErrorMsg";
+import Page from "#SRC/js/components/Page";
 
 class EditServiceModal extends Component {
   render() {
     const { id = "/" } = this.props.params;
     const serviceID = decodeURIComponent(id);
+    const serviceLoaded = DCOSStore.serviceDataReceived;
     const service = DCOSStore.serviceTree.findItemById(serviceID);
 
-    if (!service) {
+    // Loading, showing an empty modal instead
+    if (!serviceLoaded) {
       return <FullScreenModal open={true} />;
     }
 
+    // Service not found
```

Comment:
I believe the ideal way is to add a route to the `routes/service.js` with a new not found path/route at the top level and not adding conditions to low level components.

```
{
    type: Route,
    path: "*",
    component: NotFoundPage
},
```

_ps: we actually have a `NotFoundPage` component._

Reply:
I don't think it's that easy. For that we would have to move the logic whether a service is there or not into the react router. While even though we don't have a service the route is still matchable.